src/OpenLoco/src/Objects/TreeObject.cpp

```diff
@@ -109,23 +112,97 @@ namespace OpenLoco
     // 0x004BE144
     void TreeObject::load(const LoadedObjectHandle& handle, stdx::span<const std::byte> data, ObjectManager::DependentObjects*)
     {
-        Interop::registers regs;
-        regs.esi = Interop::X86Pointer(this);
-        regs.ebx = handle.id;
-        regs.ecx = enumValue(handle.type);
-        Interop::call(0x004BE144, regs);
+        auto remainingData = data.subspan(sizeof(TreeObject));
+
+        // Load localised name string
+        {
+            auto strRes = ObjectManager::loadStringTable(remainingData, handle, 0);
+            name = strRes.str;
+            remainingData = remainingData.subspan(strRes.tableLength);
+        }
+
+        // Load image offsets
+        auto imgRes = ObjectManager::loadImageTable(remainingData);
+        auto imageOffset = imgRes.imageOffset;
+        assert(remainingData.size() == imgRes.tableLength);
+
+        // Initialise sprites array
+        for (auto variant = 0; variant < 6; variant++)
+        {
+            sprites[variant] = imageOffset;
+        }
+
+        auto numVariantImages = numRotations * growth;
+        auto nextImageOffset = imageOffset;
+        for (auto variant = 0; variant < 6; variant++)
+        {
+            if ((var_3C & (1 << variant)) == 0)
+                continue;
+
+            sprites[variant] = nextImageOffset;
+            nextImageOffset += numVariantImages;
+        }
+
+        // 0x004BE186
+        auto numPrimaryImages = nextImageOffset - imageOffset;
+        nextImageOffset = imageOffset;
+
+        if ((var_3C & (1 << 5)) == 0 && (var_3C & (1 << 4)) != 0)
+        {
+            sprites[5] = sprites[4];
+        }
+
+        if ((var_3C & (1 << 5)) == 0 && (var_3C & (1 << 1)) != 0)
+        {
+            sprites[5] = sprites[1];
+        }
+
+        if ((var_3C & (1 << 4)) == 0 && (var_3C & (1 << 1)) != 0)
+        {
+            sprites[4] = sprites[1];
+        }
+
+        if ((var_3C & (1 << 1)) == 0 && (var_3C & (1 << 0)) != 0)
+        {
+            sprites[1] = sprites[0];
+        }
+
+        if ((var_3C & (1 << 0)) == 0 && (var_3C & (1 << 1)) != 0)
+        {
+            sprites[0] = sprites[1];
+        }
+
+        if ((flags & TreeObjectFlags::hasSnowVariation) != TreeObjectFlags::none)
+        {
+            for (auto variant = 0; variant < 6; variant++)
+            {
+                snowSprites[variant] = sprites[variant] + numPrimaryImages;
+            }
+
+            nextImageOffset = numPrimaryImages * 2;
+        }
+        else
+        {
+            nextImageOffset = numPrimaryImages;
+        }
+
+        if ((flags & TreeObjectFlags::hasShadow) != TreeObjectFlags::none)
+        {
+            shadowImageOffset = nextImageOffset;
+        }
```

Comment:
Could you look at adding an assert like i have for vehicle on the image offsets. Would be nice to just be sure we allocated the correct amount and have used the correct amount.